compile/enum.go

```diff
@@ -52,19 +54,26 @@ func compileEnum(file string, src *ast.Enum) (*EnumSpec, error) {
 				Reason: err,
 			}
 		}
-
 		value := prev + 1
 		if astItem.Value != nil {
 			value = *astItem.Value
 		}
 		prev = value
 
+		itemAnnotations, err := compileAnnotations(astItem.Annotations)
+		if err != nil {
+			return nil, err
```

Comment:
The intention of `compileError` is to give you a message on exactly which component failed. You should use `compileError` here and a generic `annotationConflictError{Reason: err}` (define in compile/error.go) in `compileAnnotations`. That will give us an error message like,

    cannot compile "MyEnum.SomeItem" on line 42: annotation conflict: the name "foo" has already been used on line 45